docs/worldgen/biomemodifier.md

```diff
@@ -0,0 +1,704 @@
+
+<Tabs>
+  <TabItem value="json" label="JSON" default>
+
+```json5
+{
+  "type": "neoforge:add_features",
+  // Can either be a biome id, such as "minecraft:plains",
+  // or a list of biome ids, such as ["minecraft:plains", "minecraft:badlands", ...],
+  // or a biome tag, such as "#c:is_overworld".
+  "biomes": "#namespace:your_biome_tag",
+  // Can either be a placed feature id, such as "examplemod:add_features_example",
+  // or a list of placed feature ids, such as ["examplemod:add_features_example", minecraft:ice_spike", ...],
+  // or a placed feature tag, such as "#examplemod:placed_feature_tag".
+  "features": "namespace:your_feature",
+  // See the GenerationStep.Decoration enum in code for a list of valid enum names.
+  // The decoration step section further down also has the list of values for reference.
+  "step": "underground_ores"
+}
+```
+
+  </TabItem>
+  <TabItem value="datagen" label="Datagen">
+
+```java
+// Assume we have some PlacedFeature named EXAMPLE_PLACED_FEATURE.
+// Define the ResourceKey for our BiomeModifier.
+public static final ResourceKey<BiomeModifier> ADD_FEATURES_EXAMPLE = ResourceKey.create(
+    NeoForgeRegistries.Keys.BIOME_MODIFIERS, // The registry this key is for
+    ResourceLocation.fromNamespaceAndPath(MOD_ID, "add_features_example") // The registry name
+);
+
+// BUILDER is a RegistrySetBuilder passed to DatapackBuiltinEntriesProvider
+// in a listener for GatherDataEvent.
+BUILDER.add(NeoForgeRegistries.Keys.BIOME_MODIFIERS, bootstrap -> {
+    // Lookup any necessary registries.
+    // Static registries only need to be looked up if you need to grab the tag data.
+    HolderGetter<Biome> biomes = bootstrap.lookup(Registries.BIOME);
+    HolderGetter<PlacedFeature> placedFeatures = bootstrap.lookup(Registries.PLACED_FEATURE);
+
+    // Register the biome modifiers.
+    bootstrap.register(ADD_FEATURES_EXAMPLE,
+        new AddFeaturesBiomeModifier(
+            // The biome(s) to generate within
+            HolderSet.direct(biomes.getOrThrow(Biomes.PLAINS)),
+            // The feature(s) to generate within the biomes
+            HolderSet.direct(placedFeatures.getOrThrow(EXAMPLE_PLACED_FEATURE)),
+            // The generation step
+            GenerationStep.Decoration.LOCAL_MODIFICATIONS
+        )
+    );
+})
+```
+
+  </TabItem>
+</Tabs>
+
+
+:::warning
+Care should be taken when adding vanilla `PlacedFeature`s to biomes, as doing so may cause what is known as a feature cycle violation (two biomes having the same two features in their feature lists, but in different orders within the same `GenerationStep`), leading to a crash. For similar reasons, you should not use the same `PlacedFeature` in more than one biome modifier.
+
+Vanilla `PlacedFeature`s can be referenced in biome JSONs or added via biome modifiers, but should not be used in both. If you still need to add them this way, making a copy of the vanilla `PlacedFeature` is the easiest solution to avoid these problems.
```

Comment:
```suggestion
Vanilla `PlacedFeature`s can be referenced in biome JSONs or added via biome modifiers, but should not be used in both. If you still need to add them this way, making a copy of the vanilla `PlacedFeature` under your own namespace is the easiest solution to avoid these problems.
```

Just an explicit mention to remake the feature under your namespace.